### ISSUE TITLE
why DHN?

### ISSUE THREAD

**Issue by Jacobew** [yihongXU/deepMOT#36]
Hi, thanks for your great work!
One simple question about DHN:

You claim that DHN enables end-to-end training of deep multi-object trackers, but DHN is pretrained and fixed during training. 

My question is: why not use hungarian matching? In my opinion, it always gives you the exactly right matching results. And I cannot find the comparisons of hungarian matching with DHN (from the aspects of speed/performance) in your ablations.
I've also observed that DHN (with two lstms) is much slower than Hungarian matching that runs on CPU in my project, so the speed issue may not be the reason I think. 

Please correct me if I am making mistakes here! :)

**Comment by yihongXU** [comment]
Hi, thanks for your question!

We use DHN instead of HA because we need to propagate gradients from the MOT loss. HA has operations that are not differentiable so it cannot be used directly.

Best,
